cty/function/stdlib/collection.go

```diff
@@ -563,7 +564,7 @@ var KeysFunc = function.New(&function.Spec{
 		}
 	},
 	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
-		m := args[0]
+		m, marks := args[0].UnmarkDeep()
```

Comment:
I'm surprised by the `UnmarkDeep` call here, vs shallow `Unmark`. I would have expected that we only want to unmark the top-level object, not all nested values.

As a result I don't understand how the test case where we mark a value and don't expect the mark to propagate to the resulting list is passing 🤔

Reply:
it's not, this is the wrong code, I am doing everything wrong today, so sorry